### PR TITLE
Update LIBSRC/CHAMELEON/COM submodule for linux 6.1 compilation

### LINKED ISSUE #281: Compilation in kernel 6.1 is broken
There is something in 6.1 that has changed the way some of the defined guards for VXWORK work. We get this error:

```
  CC [M]  /home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.o
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:270:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  270 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c: In function 'InitPci':
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:409:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  409 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:515:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  515 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
```

We need to fix this and also find why this only affects 6.1 and not older versions.

### PR DESCRIPTION
**Pull request by mad-jsanjuan**

### CODE REVIEW

**Review by mad-jrodriguez — APPROVED** [review]

File: MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c [hunk 1]
```diff
@@ -142,10 +142,14 @@ typedef struct smb_access_struct {
...
-static int oss_smb2_probe(struct i2c_client *client, const struct i2c_device_id *id);
+static int  oss_smb2_probe(struct i2c_client* client, const struct i2c_device_id *id);
+static int  oss_smb2_probe(struct i2c_client *client, const struct i2c_device_id *id);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+static void oss_smb2_remove(struct i2c_client *client);
+#else
 static int oss_smb2_remove(struct i2c_client *client);
-static int oss_smb2_detect(struct i2c_client *new_client, struct i2c_board_info *info);
+#endif
+static int  oss_smb2_detect(struct i2c_client *new_client, struct i2c_board_info *info);
```
Comment (mad-jrodriguez) [inline MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c]: I've seen that static int  oss_smb2_detect has 2 spaces between int and oss_smb2. I think it should be aligned with the rest of the function prototypes.

File: MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c [hunk 2]
```diff
@@ -142,10 +142,14 @@ typedef struct smb_access_struct {
 /*
  *  SMBus registering/access functions differ a bit throughout 2.4 to 2.6.x.
  */
-static int oss_smb2_probe(struct i2c_client* client, const struct i2c_device_id *id);
-static int oss_smb2_probe(struct i2c_client *client, const struct i2c_device_id *id);
+static int  oss_smb2_probe(struct i2c_client* client, const struct i2c_device_id *id);
+static int  oss_smb2_probe(struct i2c_client *client, const struct i2c_device_id *id);
```
Comment (mad-jrodriguez) [inline MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c]: oss_smb2_probe function prototype is duplicated (lines 145 and 146). Additionally, the unique change I saw is adding an extra space between the returning type (int) and the function name.

File: MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c [hunk 3]
```diff
@@ -983,8 +990,6 @@ static int oss_smb2_remove(struct i2c_client *client)
     kfree(data);
 
     /* -- data invalid now -- */
-
-    return err;
```
Comment (mad-jrodriguez) [inline MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c:987]: The returning value should be also protected with kernel version check as for kernel version < 6.1, the function oss_smb2_remove returns an integer.

**Review by mad-jrodriguez — CHANGES_REQUESTED** [review]

File: MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c (hunk 2 above)
Comment (mad-jsanjuan) [reply MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c]: How did I get to this? I'll double check

File: MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c (hunk 3 above)
Comment (mad-jsanjuan) [reply MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c:987]: This return was a bit useless as it was always returning the initial value of err = 0; I'll just leave it as a `return 0`

Comment (mad-jrodriguez) [reply MDISforLinux/DRIVERS/BBIS/SMB2BB/DRIVER/NATIVE/bb_smb2.c:987]: Protecting return err; to kernel versions below 6.1 looks good to me

**Comment by mad-jsanjuan** [comment]
Afer, revieweing the compilation logs, merging this fixes but still this one error is pending in a different pull request https://github.com/MEN-Mikro-Elektronik/13MD05-90/issues/283

**Review by mad-jrodriguez — APPROVED** [review]

**Pull request merged**